### PR TITLE
[Add Package] gotop

### PR DESCRIPTION
**Pull request by MedzikUser**

### CODE REVIEW

**Comment by xeffyr** [comment]
> `TERMUX_PKG_DESCRIPTION="A terminal based graphical activity monitor inspired by gtop and vtop"`

and it works without /proc/stat and/or other interfaces restricted by SELinux?

File: packages/gotop/build.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+TERMUX_PKG_VERSION=1:${_MAJOR_VERSION}
+TERMUX_PKG_REVISION=1
+TERMUX_PKG_SRCURL=https://github.com/xxxserxxx/gotop/archive/refs/tags/v${_MAJOR_VERSION}.tar.gz
+TERMUX_PKG_SHA256=81518fecfdab4f4c25a4713e24d9c033ba8311bbd3e2c0435ba76349028356da
+TERMUX_PKG_NO_STATICSPLIT=true
+
+termux_step_make_install() {
+	termux_setup_golang
+
+	GOOS=linux GOARCH=$TERMUX_ARCH go build -o gotop \
```
Comment (xeffyr) [inline packages/gotop/build.sh:16]: There could be only `GOOS=android`.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/gotop/build.sh (hunk 1 above)
Comment (MedzikUser) [reply packages/gotop/build.sh:16]: > > `TERMUX_PKG_DESCRIPTION="A terminal based graphical activity monitor inspired by gtop and vtop"`
> 
> and it works without `/proc/stat` or other interfaces restricted by SELinux?

Without root, there is only ram usage (same as in htop) and with root there is everything.

**Comment by MedzikUser** [comment]
> > `TERMUX_PKG_DESCRIPTION="A terminal based graphical activity monitor inspired by gtop and vtop"`
> 
> and it works without `/proc/stat` or other interfaces restricted by SELinux?

Without root, there is only ram usage (same as in htop) and with root there is everything.

**Comment by MedzikUser** [comment]
without root
ss

**Comment by MedzikUser** [comment]
with root
ss1

**Comment by MedzikUser** [comment]
htop without root
ss2

htop with root
ss3

**Comment by xeffyr** [comment]
Ok, so basically same behavior as of htop v2.x.

**Comment by MedzikUser** [comment]
> Ok, so basically same behavior as of htop v2.x.

Exactly

File: packages/gotop/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/xxxserxxx/gotop
+TERMUX_PKG_DESCRIPTION="A terminal based graphical activity monitor inspired by gtop and vtop"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@medzikuser"
+_MAJOR_VERSION=4.1.2
+# Use the ~ deb versioning construct in the future:
+TERMUX_PKG_VERSION=1:${_MAJOR_VERSION}
```
Comment (xeffyr) [inline packages/gotop/build.sh:7]: Why you are setting epoch for the package previously unavailable in Termux?

When to use epoch: https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#downgrading-the-package-or-changing-versioning-scheme

File: packages/gotop/build.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+TERMUX_PKG_REVISION=1
+TERMUX_PKG_SRCURL=https://github.com/xxxserxxx/gotop/archive/refs/tags/v${_MAJOR_VERSION}.tar.gz
+TERMUX_PKG_SHA256=81518fecfdab4f4c25a4713e24d9c033ba8311bbd3e2c0435ba76349028356da
+TERMUX_PKG_NO_STATICSPLIT=true
```
Comment (xeffyr) [inline packages/gotop/build.sh:11]: Does package provides static libraries which are for specific reasons shouldn't be splitted? If not, remove no-staticsplit var.

File: packages/gotop/build.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	termux_setup_golang
+
+	ls -alh
+
+	cd $TERMUX_PKG_SRCDIR
+
+	ls -alh
+
+	GOOS=android go build -o gotop \
+	    -ldflags "-X main.Version=v${_MAJOR_VERSION} -X main.BuildDate=$(date +%Y%m%dT%H%M%S)" \
```
Comment (xeffyr) [inline packages/gotop/build.sh:23]: You are compiling a binary but not installing it. Build system won't do this for you automatically.

File: packages/gotop/build.sh (hunk 2 above)
Comment (MedzikUser) [reply packages/gotop/build.sh:7]: I was basing it on golang and simply did not delete it but changed it to 1

File: packages/gotop/build.sh (hunk 4 above)
Comment (MedzikUser) [reply packages/gotop/build.sh:18]: Now I do not know why it does not want to compile, in my case it compiles

**Review by xeffyr — APPROVED** [review]

**Pull request merged**